libr/include/r_endian.h

```diff
@@ -0,0 +1,187 @@
+static inline ut8 read_at_ble8(const void *src, size_t offset) {
+	const ut8 *s = src;
+	s += offset;
+	return read_ble8 (s);
+}
+
+static inline void write_ble8(void *dest, ut8 val) {
+	*(ut8 *)dest = val;
+}
+
+static inline void write_at_ble8(void *dest, ut8 val, size_t offset) {
+	ut8 *d = dest + offset;
+	write_ble8 (d, val);
+}
+
+/* Big Endian functions. */
+
+static inline ut8 read_be8(const void *src) {
+	return read_ble8 (src);
+}
+
+static inline ut8 read_at_be8(const void *src, size_t offset) {
+	return read_at_ble8 (src, offset);
+}
+
+static inline void write_be8(void *dest, ut8 val) {
+	write_ble8 (dest, val);
+}
+
+static inline void write_at_be8(void *dest, ut8 val, size_t offset) {
+	write_at_ble8 (dest, val, offset);
+}
+
+static inline ut16 read_be16(const void *src) {
+	const ut8 *s = src;
+	return (((ut16)s[0]) << 8) | (((ut16)s[1]) << 0);
+}
+
+static inline ut16 read_at_be16(const void *src, size_t offset) {
+	const ut8 *s = src + offset;
+	return read_be16 (s);
+}
+
+static inline void write_be16(void *dest, ut16 val) {
+	write_be8 (dest, val >> 8);
+	write_at_be8 (dest, val >> 0, sizeof(ut8));
+}
+
+static inline void write_at_be16(void *dest, ut16 val, size_t offset) {
+	ut8 *d = dest + offset;
+	write_be16 (d, val);
+}
+
+static inline ut32 read_be32(const void *src) {
+	const ut8 *s = src;
+	return (((ut32)s[0]) << 24) | (((ut32)s[1]) << 16) |
+		(((ut32)s[2]) << 8) | (((ut32)s[3]) << 0);
+}
+
+static inline ut32 read_at_be32(const void *src, size_t offset) {
+	const ut8 *s = src + offset;
+	return read_be32 (s);
+}
+
+static inline void write_be32(void *dest, ut32 val) {
+	write_be16 (dest, val >> 16);
+	write_at_be16 (dest, val >> 0, sizeof(ut16));
+}
+
+static inline void write_at_be32(void *dest, ut32 val, size_t offset) {
+	ut8 *d = dest + offset;
+	write_be32 (d, val);
+}
+
+static inline ut64 read_be64(const void *src) {
+	ut64 val = read_be32 (src);
+	val <<= 32;
+	val |= read_at_be32 (src, sizeof(ut32));
+	return val;
+}
+
+static inline ut64 read_at_be64(const void *src, size_t offset) {
+	const ut8 *s = src + offset;
+	return read_be64 (s);
+}
+
+static inline void write_be64(void *dest, ut64 val) {
+	write_be32 (dest, val >> 32);
+	write_at_be32 (dest, val >> 0, sizeof(ut32));
+}
+
+static inline void write_at_be64(void *dest, ut64 val, size_t offset) {
+	ut8 *d = dest + offset;
+	write_be64 (d, val);
+}
+
+/* Little Endian functions. */
+
+static inline ut8 read_le8(const void *src) {
+	return read_ble8 (src);
+}
+
+static inline ut8 read_at_le8(const void *src, size_t offset) {
+	return read_at_ble8 (src, offset);
+}
+
+static inline void write_le8(void *dest, ut8 val) {
+	write_ble8 (dest, val);
+}
+
+static inline void write_at_le8(void *dest, ut8 val, size_t offset) {
+	write_at_ble8 (dest, val, offset);
+}
+
+static inline ut16 read_le16(const void *src) {
+	const ut8 *s = src;
+	return (((ut16)s[1]) << 8) | (((ut16)s[0]) << 0);
+}
+
+static inline ut16 read_at_le16(const void *src, size_t offset) {
```

Comment:
we should proabbly add a len parameter to all those function to avoid having to check for boundaries in the caller side